stablehlo/dialect/VhloTypes.td

```diff
@@ -61,9 +64,92 @@ class VHLO_TypeDef<string name,
 // VHLO Type Definitions.
 //===----------------------------------------------------------------------===//
 
-// Token type.
-def VHLO_Token : VHLO_TypeDef<"Token"> {
-  let mnemonic = "token";
+def VHLO_Token : VHLO_TypeDef<"Token", "token">;
+
+//===----------------------------------------------------------------------===//
+// Forked Types
+//===----------------------------------------------------------------------===//
+
+def VHLO_BFloat16V1 : VHLO_TypeDef<"BFloat16V1", "bf16">;
+
+def VHLO_ComplexV1 : VHLO_TypeDef<"ComplexV1", "complex", "::mlir::ComplexType"> {
+  let parameters = (ins "Type":$elementType);
+  let assemblyFormat = "`<` $elementType `>`";
+}
+
+def VHLO_Float16V1 : VHLO_TypeDef<"Float16V1", "f16">;
+
+def VHLO_Float32V1 : VHLO_TypeDef<"Float32V1", "f32">;
+
+def VHLO_Float64V1 : VHLO_TypeDef<"Float64V1","f64">;
+
+def VHLO_IndexV1 : VHLO_TypeDef<"IndexV1", "index">;
+
+def VHLO_IntegerType : VHLO_TypeDef<"IntegerV1", "integer"> {
+  let parameters = (ins "::mlir::IntegerType":$value);
+  let assemblyFormat = "`<` $value `>`";
+  let genVerifyDecl = 1;
+  let extraClassDefinition = [{
+    ::mlir::LogicalResult IntegerV1Type::verify(::llvm::function_ref<InFlightDiagnostic()> emitError,
+                     ::mlir::IntegerType value) {
+      llvm::SmallVector<unsigned> validWidths{4, 8, 16, 32, 64};
+      bool isPred = (value.getWidth() == 1 && value.isSignless());
+      if (!isPred && !llvm::is_contained(validWidths, value.getWidth())) {
+        return emitError() << "invalid integer width " << value;
+      }
+      if (!value.isSignless() && !value.isUnsigned()) {
+        return emitError() << "invalid integer signedness " << value;
+      }
+      return success();
+    }
+  }];
+}
+
+def VHLO_RankedTensorV1 : VHLO_TypeDef<"RankedTensorV1", "tensor"> {
+  let parameters = (ins
+    VHLO_Dims:$shape,
+    "::mlir::Type":$elementType,
+    "::mlir::Attribute":$encoding
+  );
+  let assemblyFormat = "`<` custom<TensorShape>($shape) `` $elementType `` custom<Encoding>($encoding) `>`";
```

Comment:
Perhaps just `Shape` (instead of `TensorShape`), given that the field is called `shape`?

Reply:
Chose the name from spec.md ebnf:

```
TensorType    ::= 'tensor' '<' TensorShape ElementType '>'
TensorShape   ::= {DimensionSize 'x'}
DimensionSize ::= digit {digit}
```

Reply:
Changed to `printShape`